src/components/network-map-tab.js

```diff
@@ -532,62 +525,65 @@ export const NetworkMapTab = ({
-            );
-        } else {
-            if (mapEquipments) {
-                console.info('Reload map equipments');
-                setWaitingLoadEquipments(true);
-                const updatedSubstationsToSend =
-                    !refIsMapManualRefreshEnabled.current &&
-                    !isUpdatedSubstationsApplied &&
-                    updatedSubstationsIds?.length > 0
-                        ? updatedSubstationsIds
-                        : undefined;
-
-                mapEquipments
-                    .reloadImpactedSubstationsEquipments(
-                        studyUuid,
-                        currentNode,
-                        updatedSubstationsToSend,
-                        setUpdatedLines
-                    )
-                    .finally(() => setWaitingLoadEquipments(false));
-                if (updatedSubstationsToSend) {
-                    setIsUpdatedSubstationsApplied(true);
-                }
+        new MapEquipments(
+            studyUuid,
+            currentNode?.id,
+            setErrorMessage,
+            dispatch,
+            intlRef
+        );
+        dispatch(resetMapReloaded());
+    }, [currentNode, dispatch, intlRef, setErrorMessage, studyUuid]);
+
+    const updateMapEquipmentsAndGeoData = useCallback(() => {
+        if (!isNodeBuilt(currentNode) || !studyUuid) {
+            return;
+        }
+        if (mapEquipments) {
+            console.info('Reload map equipments');
+            setWaitingLoadData(true);
+            const updatedSubstationsToSend =
+                !refIsMapManualRefreshEnabled.current &&
+                !isUpdatedSubstationsApplied &&
+                updatedSubstationsIds?.length > 0
+                    ? updatedSubstationsIds
+                    : undefined;
+
+            mapEquipments
+                .reloadImpactedSubstationsEquipments(
+                    studyUuid,
+                    currentNode,
+                    updatedSubstationsToSend,
+                    setUpdatedLines
+                )
+                .then(loadGeoData)
+                .finally(() => setWaitingLoadData(false));
+            if (updatedSubstationsToSend) {
+                setIsUpdatedSubstationsApplied(true);
             }
         }
         dispatch(resetMapReloaded());
     }, [
         currentNode,
         dispatch,
-        intlRef,
-        isInitialized,
         isUpdatedSubstationsApplied,
         mapEquipments,
-        setErrorMessage,
         studyUuid,
         updatedSubstationsIds,
+        loadGeoData,
     ]);
 
     useEffect(() => {
-        if (studyUpdatedForce.eventData.headers) {
+        if (isInitialized && studyUpdatedForce.eventData.headers) {
             if (
                 studyUpdatedForce.eventData.headers[UPDATE_TYPE_HEADER] ===
                 'loadflow'
             ) {
                 //TODO reload data more intelligently
-                loadMapEquipments();
+                updateMapEquipmentsAndGeoData();
```

Comment:
We need to update geo data after a loadflow ?

Reply:
I just kept the old behaviour (after updating map equipment we always called 'loadGeoData'), so do I change this ?

Reply:
Yes it's useless to avoid calculating that no geodata is missing here and only load map equipments

Reply:
I refactor a bit